File: 박인우/[프로그래머스] 행렬 테두리 회전하기.py
def solution(rows, columns, queries):
    answer = []
    matrix = [[i*columns + j
               for j in range(1, columns+1)]for i in range(0, rows)]
    for querie in queries:
        minNum = 10001
        fx, fy, tx, ty = map(lambda x: x-1, querie)
        lt = matrix[fx][fy]
        rt = matrix[fx][ty]
        lb = matrix[tx][fy]
        rb = matrix[tx][ty]
        minNum = min(minNum, lt)
        minNum = min(minNum, rt)
        minNum = min(minNum, lb)
        minNum = min(minNum, rb)
        for i in range(ty-1, fy, -1):
            minNum = min(minNum, matrix[fx][i])
            matrix[fx][i+1] = matrix[fx][i]
        for i in range(tx-1, fx, -1):
            minNum = min(minNum, matrix[i][ty])
            matrix[i+1][ty] = matrix[i][ty]
        for i in range(fy+1, ty, 1):
            minNum = min(minNum, matrix[tx][i])
            matrix[tx][i-1] = matrix[tx][i]
        for i in range(fx+1, tx, 1):
            minNum = min(minNum, matrix[i][fy])
            matrix[i-1][fy] = matrix[i][fy]
        matrix[fx][fy+1] = lt
        matrix[fx+1][ty] = rt
        matrix[tx-1][fy] = lb
        matrix[tx][ty-1] = rb
        answer.append(minNum)
    return answer

File: 박인우/0207-0213/[프로그래머스] 행렬 테두리 회전하기.swift
import Foundation

func solution(_ rows:Int, _ columns:Int, _ queries:[[Int]]) -> [Int] {
    var matrix = [[Int]]()
    var answer = [Int]()
    
    for i in 0..<rows {
        var temp = [Int]()
        for j in 1...columns {
            temp.append(columns*i + j)
        }
        matrix.append(temp)
    }
    
    for query in queries {
        let x1 = query[0]-1
        let x2 = query[2]-1
        let y1 = query[1]-1
        let y2 = query[3]-1
        
        var list = [Int]()
        var minValue = matrix[x1+1][y1]
        var store = matrix[x1+1][y1]
        
        for y in y1...y2 {
            var temp = store
            store = matrix[x1][y]
            matrix[x1][y] = temp
            minValue = min(minValue, store)
        }
        
        for x in x1+1...x2 {
            var temp = store
            store = matrix[x][y2]
            matrix[x][y2] = temp
            minValue = min(minValue, store)
        }
        
        for y in stride(from: y2-1, through: y1, by: -1) {
            var temp = store
            store = matrix[x2][y]
            matrix[x2][y] = temp
            minValue = min(minValue, store)
        }
        
        for x in stride(from: x2-1, through: x1+1, by: -1) {
            var temp = store
            store = matrix[x][y1]
            matrix[x][y1] = temp
            minValue = min(minValue, store)
        }
        
        answer.append(minValue)
    }
    
    return answer
}
